exdb/views.py

```diff
@@ -107,34 +109,52 @@ def get_success_url(self):
 
     def get_context_data(self, **kwargs):
         context = super(ExperienceApprovalView, self).get_context_data()
-        context['experience'] = self.get_experience()
+        if kwargs.get('invalid_comment'):
+            context['comment_form'] = kwargs.get('invalid_comment')
+        else:
+            context['comment_form'] = self.second_form_class()
         context['comments'] = ExperienceComment.objects.filter(experience=context['experience'])
         return context
 
-    def form_valid(self, form):
-        form.instance.author = self.request.user
-        form.instance.experience = self.get_experience()
-        form.instance.experience.status = 'ad' if self.request.POST.get('approve') else 'de'
-        if form.instance.experience.status == 'ad':
-            form.instance.experience.next_approver = None
-            ExperienceApproval.objects.create(
-                experience=form.instance.experience,
-                approver=self.request.user
-            )
-        form.instance.experience.save()
-        return super(ExperienceApprovalView, self).form_valid(form)
-
-    def form_invalid(self, form):
+    def post(self, request, *args, **kwargs):
+        self.object = self.get_object()
+        experience_form = self.get_form()
+        comment_form = self.second_form_class(request.POST)
+        if experience_form.is_valid() and comment_form.is_valid():
+            return self.form_valid(experience_form, comment_form)
+        else:
+            return self.form_invalid(experience_form, comment_form)
+
+    def form_valid(self, experience_form, comment_form):
+        comment_form.instance.author = self.request.user
+        experience_form.instance.author = self.get_object().author
```

Comment:
It would seem not anymore.  Much has changed since that line was added, but at one point the form was not validating because it wasn't getting the experience's author.  I just forgot to test it without that line since it changed.